src/libraries/System.Private.CoreLib/src/System/Reflection/FieldAccessor.cs

```diff
@@ -0,0 +1,534 @@
+                        Volatile.Write(
+                            ref Unsafe.AsRef<short>(_addressOrOffset.ToPointer()),
+                            Unsafe.As<byte, short>(ref value!.GetRawData()));
+                        return;
+
+                    case FieldAccessorType.StaticValueTypeSize4:
+                        VerifyStaticField(ref value, invokeAttr, binder, culture);
+                        Volatile.Write(
+                            ref Unsafe.AsRef<int>(_addressOrOffset.ToPointer()),
+                            Unsafe.As<byte, int>(ref value!.GetRawData()));
+                        return;
+
+                    case FieldAccessorType.StaticValueTypeSize8:
+                        VerifyStaticField(ref value, invokeAttr, binder, culture);
+                        Volatile.Write(
+                            ref Unsafe.AsRef<long>(_addressOrOffset.ToPointer()),
+                            Unsafe.As<byte, long>(ref value!.GetRawData()));
+                        return;
+
+                    case FieldAccessorType.SlowPathUntilClassInitialized:
+                        {
+                            if (IsStatic())
+                            {
+                                VerifyStaticField(ref value, invokeAttr, binder, culture);
+                            }
+                            else
+                            {
+                                VerifyInstanceField(obj, ref value, invokeAttr, binder, culture);
+                            }
+
+                            isClassInitialized = false;
+                            RuntimeFieldHandle.SetValue(_fieldInfo, obj, value, (RuntimeType)_fieldInfo.FieldType, _fieldInfo.m_declaringType, ref isClassInitialized);
+                            if (isClassInitialized)
+                            {
+                                Initialize();
+                            }
+
+                            return;
+                        }
+
+                    case FieldAccessorType.NoInvoke:
+                        if (_fieldInfo.DeclaringType is not null && _fieldInfo.DeclaringType.ContainsGenericParameters)
+                            throw new InvalidOperationException(SR.Arg_UnboundGenField);
+
+                        throw new FieldAccessException();
+                }
+            }
+
+            // Slow path
+            if (IsStatic())
+            {
+                VerifyStaticField(ref value, invokeAttr, binder, culture);
+            }
+            else
+            {
+                VerifyInstanceField(obj, ref value, invokeAttr, binder, culture);
+            }
+
+            isClassInitialized = true;
+            RuntimeFieldHandle.SetValue(_fieldInfo, obj, value, (RuntimeType)_fieldInfo.FieldType, _fieldInfo.m_declaringType, ref isClassInitialized);
+        }
+
+        private void InitializeClass()
+        {
+            if (_fieldInfo.DeclaringType is null)
```

Comment:
Why is the `isClassInitialized` flag not enough to deal with the initialization? This looks redundant.

Reply:
I was able to remove this, but I needed to remove the "readonly" from a couple fields.